File: src/main/java/com/bopr/intellij/iipowershell/run/PowerShellRunConfigurationOptions.java
package com.bopr.intellij.iipowershell.run;

import com.intellij.execution.configurations.RunConfigurationOptions;
import com.intellij.openapi.components.StoredProperty;

public class PowerShellRunConfigurationOptions extends RunConfigurationOptions {

    private final StoredProperty<String> scriptPath = string("").provideDelegate(this, "scriptPath");
    private final StoredProperty<String> scriptArguments = string("").provideDelegate(this, "scriptArguments");
    private final StoredProperty<String> executablePath = string("pwsh.exe").provideDelegate(this, "executablePath");

    public String getScriptPath() {
        return scriptPath.getValue(this);
    }

    public void setScriptPath(String value) {
        scriptPath.setValue(this, value);
    }

    public String getScriptArguments() {
        return scriptArguments.getValue(this);
    }

    public void setScriptArguments(String value) {
        scriptArguments.setValue(this, value);
    }

    public String getExecutablePath() {
        return executablePath.getValue(this);
    }

    public void setExecutablePath(String value) {
        executablePath.setValue(this, value);
    }

}

File: src/main/kotlin/com/bopr/intellij/iipowershell/run/PowerShellRunConfigurationOptions.kt
package com.bopr.intellij.iipowershell.run

import com.intellij.execution.configurations.RunConfigurationOptions
import com.intellij.openapi.components.StoredProperty

class PowerShellRunConfigurationOptions : RunConfigurationOptions() {

    private val scriptPathProperty: StoredProperty<String?> = string("").provideDelegate(this, "scriptPath")
    private val scriptArgumentsProperty: StoredProperty<String?> = string("").provideDelegate(this, "scriptArguments")
    private val executablePathProperty: StoredProperty<String?> =
        string("pwsh.exe").provideDelegate(this, "executablePath")

    var scriptPath
        get() = scriptPathProperty.getValue(this)!!
        set(value) = scriptPathProperty.setValue(this, value)

    var scriptArguments
        get() = scriptArgumentsProperty.getValue(this)
        set(value) = scriptArgumentsProperty.setValue(this, value)

    var executablePath: String
        get() = executablePathProperty.getValue(this)!!
        set(value) = executablePathProperty.setValue(this, value)
}